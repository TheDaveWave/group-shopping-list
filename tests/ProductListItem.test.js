import React from "react";
import renderer from "react-test-renderer";
import ProductListItem from "../src/components/ProductList/ProductListItem";

const getProducts = () => {
  axios
    .get("/products")
    .then((response) => {
      console.log(response.data);
      setProductList(response.data);
    })
    .catch((err) => {
      if (err.response.status === 404) {
        return;
      } else {
        console.log("error in axios GET", err);
      }
    });
};

const product = {
    id: 5,
    name: "Balls",
    quantity: 2,
    unit: "balls",
    isPurchased: false,
}

test("Renders Product Item", () => {
  const tree = renderer.create(<ProductListItem product={product} getProducts={getProducts}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
