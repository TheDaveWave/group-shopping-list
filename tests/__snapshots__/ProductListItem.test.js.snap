// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Product Item 1`] = `
<div
  className="item-box"
>
  <p>
    Balls
  </p>
  <p>
    2
     
    balls
  </p>
  <div
    className="btn-box"
  >
    <button
      className="list-btn"
      onClick={[Function]}
    >
      âœ“
    </button>
    <button
      className="list-btn"
      onClick={[Function]}
    >
      X
    </button>
  </div>
</div>
`;
